### PR TITLE
Make iPad usage comfortable

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
- Add font size increase/decrease buttons
- Add chapter select for smaller texts 

image

## Related Issue

- Fixes https://github.com/owncloud/web/issues/10464

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by JanAckermann** [comment]
I also increased the left side bar chapter navigation paddings and text sizes again, because it looked somewhat messy, please give feedback

File: packages/web-app-epub-reader/src/App.vue [hunk 1]
```diff
@@ -163,13 +222,27 @@ export default defineComponent({
...
+      FONT_SIZE_PERCENTAGE_STEP
     }
   }
 })
 </script>
 <style lang="scss">
 .epub-reader {
+  &-controls {
+    &-chapters-select {
+      @media (min-width: $oc-breakpoint-large-default) {
```
Comment (kulmann) [inline packages/web-app-epub-reader/src/App.vue:240]: we already have a class for that: `oc-hidden@l`

**Review by kulmann — CHANGES_REQUESTED** [review]
Having a small css nag, see comment 😁 

I had the feeling that I would like to know the current font size percentage if != 100%. Currently it's only possible to deduct the current percentage from the tooltips on the two buttons. Do you have a good idea for that maybe?

File: packages/web-app-epub-reader/src/App.vue [hunk 2]
```diff
@@ -44,6 +84,7 @@ import {
 } from '@ownclouders/web-pkg'
 import ePub, { Book, NavItem, Rendition, Location } from 'epubjs'
 import { DisplayedLocation } from 'epubjs/types/rendition'
+import { $gettext } from '@ownclouders/web-pkg/src/router/utils'
```
Comment (JanAckermann) [inline packages/web-app-epub-reader/src/App.vue:87]: 🤬🤬🤬

File: packages/web-app-epub-reader/src/App.vue [hunk 3]
```diff
@@ -79,6 +122,7 @@ export default defineComponent({
     const currentChapter = ref<NavItem>()
     const navigateLeftDisabled = ref(true)
     const navigateRightDisabled = ref(false)
+    const fontSizePercentage = ref(100)
```
Comment (JanAckermann) [inline packages/web-app-epub-reader/src/App.vue:125]: current...

File: packages/web-app-epub-reader/src/App.vue [hunk 4]
```diff
@@ -4,36 +4,76 @@
...
+            class="epub-reader-controls-font-size-reset"
+            gap-size="none"
+            @click="resetFontSize"
+            v-text="`${fontSizePercentage}%`"
+          />
+          <oc-button
+            v-oc-tooltip="`${fontSizePercentage + FONT_SIZE_PERCENTAGE_STEP}%`"
+            class="epub-reader-controls-font-size-increare"
+            :disabled="increaseFontSizeDisabled"
+            gap-size="none"
```
Comment (JanAckermann) [inline packages/web-app-epub-reader/src/App.vue:43]: Rm ?

File: packages/web-app-epub-reader/src/App.vue [hunk 5]
```diff
@@ -163,9 +237,17 @@ export default defineComponent({
...
+      increaseFontSize,
+      decreaseFontSize,
+      increaseFontSizeDisabled,
+      decreaseFontSizeDisabled,
+      fontSizePercentage,
+      FONT_SIZE_PERCENTAGE_STEP
     }
-  }
+  },
+  methods: { $gettext }
```
Comment (kulmann) [inline packages/web-app-epub-reader/src/App.vue:250]: No options API please. Why do you need it at all? When I remove it my IDE still shows $gettext correctly in the template...

File: packages/web-app-epub-reader/src/App.vue (hunk 4 above)
Comment (kulmann) [reply packages/web-app-epub-reader/src/App.vue:43]: looks fine like it is now.

File: packages/web-app-epub-reader/src/App.vue [hunk 6]
```diff
@@ -4,36 +4,76 @@
...
       </li>
     </oc-list>
-    <div class="oc-flex oc-flex-center oc-width-1-1">
-      <div class="oc-flex oc-flex-middle oc-mx-l">
-        <oc-button :disabled="navigateLeftDisabled" appearance="raw" @click="navigateLeft">
-          <oc-icon name="arrow-left-s" fill-type="line" size="xlarge" />
-        </oc-button>
+    <div class="oc-width-1-1 oc-height-1-1">
+      <div class="epub-reader-controls oc-flex oc-flex-middle oc-m-s">
+        <div class="oc-flex oc-button-group">
```
Comment (kulmann) [inline packages/web-app-epub-reader/src/App.vue:21]: Button group breaks into two lines. Also, is it intended that the dropdown spans the full width?

Screenshot 2024-02-20 at 06 29 59

File: packages/web-app-epub-reader/src/App.vue [hunk 7]
```diff
@@ -4,36 +4,76 @@
...
+            gap-size="none"
+            @click="increaseFontSize"
+          >
+            <oc-icon name="font-family" fill-type="none" size="small" />
+            <oc-icon name="add" size="xsmall" />
+          </oc-button>
+        </div>
+        <oc-select
+          v-model="currentChapter"
+          class="epub-reader-controls-chapters-select oc-width-1-1 oc-px-s oc-hidden@l"
```
Comment (kulmann) [inline packages/web-app-epub-reader/src/App.vue:52]: Don't know why, but I don't have a `pointer` cursor when hovering over the OcSelect. Could you add the `oc-cursor-pointer` class and check if that helps?

**Review by kulmann — CHANGES_REQUESTED** [review]
I like where this is going, really nice improvements! 👨‍🎨 

Left a few comments, nothing big.

But I do have one wish: Could you please add some unit tests? Buttons for font size reset and navigating back/forth should be easy to do. Chapter selection would make sense as well. And that small and big screen sizes show a different chapter selection. Happy to have that in a followup!

File: packages/web-app-epub-reader/src/App.vue (hunk 5 above)
Comment (JanAckermann) [reply packages/web-app-epub-reader/src/App.vue:250]: I don't need it at all, I marked a comment already. My ide is always auto importing it 🤬

File: packages/web-app-epub-reader/src/App.vue (hunk 6 above)
Comment (JanAckermann) [reply packages/web-app-epub-reader/src/App.vue:21]: I think a big select helps on mobile view (iPad) what do you think ? Better on the initial screenshot ?

**Comment by JanAckermann** [comment]
Yes! I planned unit tests and a changelog item in a follow up

File: packages/web-app-epub-reader/src/App.vue (hunk 7 above)
Comment (JanAckermann) [reply packages/web-app-epub-reader/src/App.vue:52]: Could you compare to our other oc selects ? I don't see a difference (even if cursor is not a pointer but smth else)

File: packages/web-app-epub-reader/src/App.vue (hunk 5 above)
Comment (kulmann) [reply packages/web-app-epub-reader/src/App.vue:250]: ah, ok, so only a reminder for yourself 👍

File: packages/web-app-epub-reader/src/App.vue (hunk 6 above)
Comment (kulmann) [reply packages/web-app-epub-reader/src/App.vue:21]: hmm... don't have a hard preference and on even smaller screen it makes sense to use the full width. so maybe best to just keep it like this and wait for feedback.

File: packages/web-app-epub-reader/src/App.vue (hunk 7 above)
Comment (kulmann) [reply packages/web-app-epub-reader/src/App.vue:52]: Aaaah, found it. The `:searchable="false"` attribute is missing. :-) Please add it, no free text input allowed for this dropdown, only selecting the predefined options is allowed.

**Review by kulmann — APPROVED** [review]
Really, really awesome! 😍 

One last thing: please store the font size in the localStorage as well. 😇

File: packages/web-app-epub-reader/src/App.vue [hunk 8]
```diff
@@ -227,6 +232,10 @@ export default defineComponent({
 
     watch(currentFontSizePercentage, () => {
       rendition.themes.fontSize(`${unref(currentFontSizePercentage)}%`)
+      localStorageData.value = {
+        ...localStorageData,
```
Comment (kulmann) [inline packages/web-app-epub-reader/src/App.vue:236]: Needs to be `...unref(localStorageData)`

File: packages/web-app-epub-reader/src/App.vue [hunk 9]
```diff
@@ -155,6 +230,14 @@ export default defineComponent({
       }
     )
 
+    watch(currentFontSizePercentage, () => {
+      rendition.themes.fontSize(`${unref(currentFontSizePercentage)}%`)
+      localStorageData.value = {
+        ...localStorageData,
```
Comment (JanAckermann) [inline packages/web-app-epub-reader/src/App.vue:236]:
```suggestion
        ...unref(localStorageData),
```

**Pull request merged**